### PR TITLE
Add example for loading a LightningModule if it has additional init arguments

### PR DESCRIPTION
**Pull request by rasbt**
## What does this PR do?

It clarifies how to use checkpoints  if LightningModules have additional init arguments

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18838.org.readthedocs.build/en/18838/

### CODE REVIEW

File: docs/source-pytorch/common/checkpointing_basic.rst [hunk 1]
```diff
@@ -122,6 +122,20 @@ If you used the *self.save_hyperparameters()* method in the init of the Lightnin
     # uses in_dim=128, out_dim=10
     model = LitModel.load_from_checkpoint(PATH, in_dim=128, out_dim=10)
 
+Similarly, if you used other positional arguments or keyword arguments in the *__init__* method of the LightningModule,
```
Comment (Borda) [inline docs/source-pytorch/common/checkpointing_basic.rst:125]:
```suggestion
Similarly, if you used other positional arguments or keyword arguments in the ``__init__`` method of the LightningModule,
```

**Review by Borda — APPROVED** [review]

**Comment by rasbt** [comment]
@Borda I accepted your suggestion because I was also in favor of using code formatting instead of `*` for `__init__`, but on that note, I saw that * was used for self.save_hyperparameters() etc.

File: docs/source-pytorch/common/checkpointing_basic.rst [hunk 2]
```diff
@@ -122,6 +122,20 @@ If you used the *self.save_hyperparameters()* method in the init of the Lightnin
     # uses in_dim=128, out_dim=10
     model = LitModel.load_from_checkpoint(PATH, in_dim=128, out_dim=10)
 
+Similarly, if you used other positional arguments or keyword arguments in the ``__init__`` method of the LightningModule,
+you can initialize the model as follows:
```
Comment (awaelchli) [inline docs/source-pytorch/common/checkpointing_basic.rst:126]: There is still one missing concept here that we didn't explain, which I think would resolve some confusion.
We've explained that

1) if you call `save_hyperparameters()`, the init params get saved and then when you load using `load_from_checkpoint`, you can override them if you want but it's optional.


But we haven't explained

2) If you didn't call `save_hyperparameters()` OR you excluded some parameters to be saved via `save_hyperparameters(ignore=...)` then you MUST pass the missing arguments when calling `load_from_checkpoint`. This is precisely what was needed in the example that @rasbt discovered. 

So I would use this opportunity to extend the sentence above to explain that.

**Review by awaelchli — APPROVED** [review]
Thanks!

File: docs/source-pytorch/common/checkpointing_basic.rst (hunk 2 above)
Comment (rasbt) [reply docs/source-pytorch/common/checkpointing_basic.rst]: Thanks @awaelchli . I extended the explanation a bit to clarify.

**Review by awaelchli — APPROVED** [review]

**Comment by awaelchli** [comment]
Great! Thanks for the update

**Pull request merged**